src/test/java/net/marcellperger/mathexpr/parser/ParserTest.java

```diff
@@ -3,180 +3,229 @@
+            assertInfixParsesTo(CommonData.getBigData3Pow_minimumParens(), ADD_PREC);
+            assertInfixParsesTo(CommonData.getBigData3Pow_groupingParens(), ADD_PREC);
+        }
     }
 
-    @Test
-    void parsePrecedenceLevel_nocache() {
-        boolean origNocache = nocache;
-        nocache = true;
-        try {
-            parsePrecedenceLevel();
-        } finally {
-            nocache = origNocache;
+    @ParameterizedTest
+    @ValueSource(booleans={true, false})
+    void parse_pow(boolean disableCache) {
+        try(var ignored = setNocacheAttr(disableCache)) {
+            assertParsesTo("1.2**9.1",
+                new PowOperation(new BasicDoubleSymbol(1.2), new BasicDoubleSymbol(9.1)));
+            assertParsesTo("1.2**9.1**.3",
+                new PowOperation(new BasicDoubleSymbol(1.2), new PowOperation(new BasicDoubleSymbol(9.1), new BasicDoubleSymbol(.3))));
+            assertParsesTo("1.2**9.1+.3",
+                new AddOperation(new PowOperation(new BasicDoubleSymbol(1.2), new BasicDoubleSymbol(9.1)), new BasicDoubleSymbol(.3)));
+            assertParsesTo(CommonData.getBigData3Pow_minimumParens());
+            assertParsesTo(CommonData.getBigData3Pow_groupingParens());
         }
     }
 
-    static class WithSuppressingCache implements AutoCloseable {
-        public static WithSuppressingCache getInstance() {
-            return INSTANCE;
+    protected WithNocache setNocacheAttr(boolean disableCache) {
+        return new WithNocache(this, disableCache);
+    }
+
+    static class WithNocache implements AutoCloseable {
+        boolean origNocache;
+        boolean doDisable;
+        ParserTest inst;
+
+        public WithNocache(ParserTest inst_) {
+            this(inst_, true);
+        }
+        public WithNocache(ParserTest inst_, boolean doDisable_) {
+            inst = inst_;
+            doDisable = doDisable_;
+            _start();
         }
-        protected static WithSuppressingCache INSTANCE = new WithSuppressingCache();
-        protected WithSuppressingCache() {}
 
-        WithSuppressingCache startInstance() {
-            clearCache();
-            return this;
+        private void _start() {
+            origNocache = inst.nocache;
+            inst.nocache = doDisable;
         }
-        static WithSuppressingCache start() {
-            return getInstance().startInstance();
+
+        public void close() {
+            inst.nocache = origNocache;
         }
-        WithSuppressingCache stop() {
-            restoreCache();
-            return this;
+    }
+
+    protected WithSuppressingCache applyNocacheAttr() {
+        return WithSuppressingCache.start(nocache);
+    }
+
+    /**
+     *
+     * Public API:
+     * <ul>
+     *     <li>Create an instance and clear cache: {@link #start(boolean doStart)} / {@link #start()}</li>
+     *     <li>Restore the cache: {@link #close()}</li>
+     * </ul>
+     */
+    static class WithSuppressingCache implements AutoCloseable {
```

Comment:
Is this JavaDoc comment really necessary - it's just an internal thing and it should e clear enough from `public` / `protected`